components/motor/gpio/motor_encoder.go

```diff
@@ -354,184 +240,162 @@ func (m *EncodedMotor) rpmMonitor() {
+		m.state.lastPowerPct = powerPct
+		if err := m.setPower(m.cancelCtx, powerPct, true); err != nil {
+			panic(err)
 		}
-		return false
-	}
-
-	// slow down so we don't overshoot
-	// halve and quarter rpm values based on seconds remaining in move
-
-	desiredRPM := m.state.desiredRPM
-	timeLeftSeconds := math.Abs(60.0 * rotationsLeft / desiredRPM)
-
-	desiredRPM = slowDownMath(timeLeftSeconds, desiredRPM, m.rampRate)
-
-	if rpmDebug {
-		m.logger.Debugf(" - rotationsLeft %.2f timeLeftSeconds %.2f rpm(%v -> %v)",
-			rotationsLeft, timeLeftSeconds, m.state.desiredRPM, desiredRPM)
-	}
-
-	m.rpmMonitorPassSetRpmInLock(currentRPM, desiredRPM, rotationsLeft, rpmDebug)
-
-	return true
-}
-
-// TODO(erh): someone make this better.
-func slowDownMath(timeLeftSeconds, desiredRPM, rampRate float64) float64 {
-	if timeLeftSeconds <= 0 {
-		return desiredRPM
-	}
-
-	if timeLeftSeconds < .5 {
-		desiredRPM *= math.Min(1, rampRate/.5)
+		return
 	}
+	m.logger.Debugf("currentRPM = %v, desiredRPM = %v", m.state.currentRPM, m.state.desiredRPM)
 
-	if timeLeftSeconds < .2 {
-		desiredRPM *= .5
+	if (dir == 1 && m.state.currentRPM > m.state.desiredRPM) || (dir == -1 && m.state.currentRPM < m.state.desiredRPM) {
+		powerPct := m.state.lastPowerPct - (m.rampRate * m.directionMovingInLock())
+		if sign(powerPct) != m.directionMovingInLock() {
+			powerPct = m.state.lastPowerPct
+		}
+		if err := m.setPower(m.cancelCtx, powerPct, true); err != nil {
+			panic(err)
+		}
 	}
-
-	if timeLeftSeconds < .075 {
-		desiredRPM *= .5 * math.Max(1, (1.5-rampRate))
+	if (dir == 1 && m.state.currentRPM <= m.state.desiredRPM) || (dir == -1 && m.state.currentRPM >= m.state.desiredRPM) {
+		powerPct := m.state.lastPowerPct + (m.rampRate * m.directionMovingInLock())
+		if sign(powerPct) != m.directionMovingInLock() {
+			powerPct = m.state.lastPowerPct
+		}
+		if err := m.setPower(m.cancelCtx, powerPct, true); err != nil {
+			panic(err)
+		}
 	}
-
-	return desiredRPM
 }
 
-func (m *EncodedMotor) computeRPM(pos, lastPos, now, lastTime int64) float64 {
-	minutes := float64(now-lastTime) / (1e9 * 60)
-	if minutes == 0 {
-		return 0.0
-	}
-	rotations := float64(pos-lastPos) / float64(m.ticksPerRotation)
-	return rotations / minutes
+// RPMMonitorCalls returns the number of calls RPM monitor has made.
+func (m *EncodedMotor) RPMMonitorCalls() int64 {
+	return atomic.LoadInt64(&m.rpmMonitorCalls)
 }
 
-func (m *EncodedMotor) computeNewPowerPct(currentRPM, desiredRPM float64) float64 {
-	lastPowerPct := m.state.lastPowerPct
-
-	if desiredRPM*currentRPM < 0 {
-		// if desiredRPM and currentRPM are different signs, we're going the wrong direction
-		// treat this as if we're not moving, so we can increase power because going the wrong direction
-		// is almost worse than not moving.
-		currentRPM = 0
-	}
+// IsRegulated returns if the motor is currently regulated or not.
+func (m *EncodedMotor) IsRegulated() bool {
```

Comment:
Are we using these two functions anywhere? I think they'd be good to use in a future pr where we get the state variables alla t once.

Reply:
no they aren't being used (and they weren't before either) so I assumed they were just for SDK purposes? but I will remove for now and add a ticket to the epic